bansa/src/main/kotlin/com/brianegan/bansa/RxStore.kt

```diff
@@ -1,31 +1,24 @@
 package com.brianegan.bansa
 
 import rx.Observable
-import rx.Scheduler
 import rx.Subscriber
 import rx.Subscription
-import rx.schedulers.Schedulers
 import rx.subjects.PublishSubject
 import rx.subjects.SerializedSubject
 
 class RxStore<S, A>(
-        private val initialState: S,
-        private val initialReducer: (S, A) -> S,
-        private val scheduler: Scheduler = Schedulers.newThread()
+        override var state: S,
+        val reducer: (S, A) -> S
 ) : Store<S, A> {
-    private val dispatcher: SerializedSubject<A, A>
-    override var state: S = initialState
+
+    private val dispatcher = SerializedSubject<A, A>(PublishSubject.create<A>())
 
     override val stateChanges: Observable<S>
-    var reducer: (S, A) -> S
 
     init {
-        reducer = initialReducer
-        dispatcher = SerializedSubject<A, A>(PublishSubject.create<A>())
         stateChanges = dispatcher // When an action is dispatched
-                .observeOn(scheduler) // Run the scan on a given thread, by default a background thread
```

Comment:
First, I checked about the observeOn vs. subscribeOn, you were right, using observeOn is the only to run reducers on a specific scheduler.

Then I thought about it, I think that expensive computation should never be done on the reducer level. If there is anything expensive to be done, this should be handled the same as we handle async requests, that is to say by making expensive computation outside the event stream (maybe using a thunk middleware). That's why I would prefer not making it possible by not allowing specifying a scheduler. Also, I'm not a redux of Flux expert, but I think that synchronicity of the stream is one of the core principle of these frameworks. One should be able to dispatch and action, and then query the state and observe changes made by the action without worry about asynchronicity of reducers.